mod/youtubevideo/renderer.php

```diff
@@ -57,18 +57,37 @@ public function module_intro($intro)
      */
     public function youtube_video($youtube_id)
     {
-        $video_container = html_writer::start_div('youtubevideo-container');
-        $video_container .= html_writer::empty_tag('iframe', array(
-            'width' => '560',
-            'height' => '315',
-            'src' => "https://www.youtube.com/embed/$youtube_id",
-            'frameborder' => '0',
-            'allow' => 'autoplay; encrypted-media',
-            'allowfullscreen' => 'allowfullscreen'
-        ));
-        $video_container .= html_writer::end_div();
-
-        return $video_container;
+        $thumbnail_url = "https://img.youtube.com/vi/{$youtube_id}/0.jpg";
+        $embed_url = "https://www.youtube-nocookie.com/embed/{$youtube_id}";
+
+        $output = html_writer::start_div('youtubevideo-container', ['data-youtube-id' => $youtube_id]);
+        $output .= html_writer::empty_tag('img', [
+            'src' => $thumbnail_url,
+            'alt' => 'Video thumbnail',
+            'class' => 'youtube-thumbnail'
+        ]);
+        $output .= html_writer::div('', 'play-button');
+        $output .= html_writer::end_div();
+
+        $this->page->requires->js_amd_inline("
+        require(['jquery'], function($) {
+            $('.youtubevideo-container').on('click', function() {
+                var container = $(this);
+                var youtubeId = container.data('youtube-id');
+                var iframe = $('<iframe>', {
+                    src: '{$embed_url}?autoplay=1',
+                    frameborder: '0',
+                    allow: 'autoplay; encrypted-media',
+                    allowfullscreen: 'allowfullscreen',
+                    width: '100%',
+                    height: '100%'
+                });
+                container.html(iframe);
+            });
+        });
+    ");
```

Comment:
Cette façon de faire de l'inline js est dangereuse et coûteuse à terme pour la maintenance.

Reply:
Tu peux procéder comme ça.
```suggestion
define([], function() {
    return {
        init: function() {
            document.querySelectorAll('.youtubevideo-container').forEach(function(container) {
                container.addEventListener('click', function() {
                    var youtubeId = container.getAttribute('data-youtube-id');
                    var embedUrl = 'https://www.youtube-nocookie.com/embed/' + youtubeId + '?autoplay=1';
                    var iframe = document.createElement('iframe');
                    iframe.setAttribute('src', embedUrl);
                    iframe.setAttribute('frameborder', '0');
                    iframe.setAttribute('allow', 'autoplay; encrypted-media');
                    iframe.setAttribute('allowfullscreen', '');
                    iframe.style.width = '100%';
                    iframe.style.height = '100%';
                    container.innerHTML = '';
                    container.appendChild(iframe);
                });
            });
        }
    };
});
```

Reply:
D'accord j'essaie ton idée.